pkg/license/license.go

```diff
@@ -37,29 +37,76 @@ const (
 	Type = "elastic-usage"
 	// GiB represents the number of bytes for 1 GiB
 	GiB = 1024 * 1024 * 1024
+
+	elasticsearchKey = "elasticsearch"
+	kibanaKey        = "kibana"
+	apmKey           = "apm"
+	entSearchKey     = "enterprise_search"
+	logstashKey      = "logstash"
+	totalKey         = "total_managed"
 )
 
+type managedMemory struct {
+	resource.Quantity
+	label string
+}
+
+func newManagedMemory(binarySI int64, label string) managedMemory {
+	return managedMemory{
+		Quantity: *resource.NewQuantity(binarySI, resource.BinarySI),
+		label:    label,
+	}
+}
+
+func (mm managedMemory) inGiB() float64 {
+	return inGiB(mm.Quantity)
+}
+
+func (mm managedMemory) intoMap(m map[string]string) {
+	m[mm.label+"_memory"] = fmt.Sprintf("%0.2fGiB", inGiB(mm.Quantity))
+	m[mm.label+"_memory_bytes"] = fmt.Sprintf("%d", mm.Quantity.Value())
+}
+
+type memoryUsage struct {
+	appUsage    map[string]managedMemory
+	totalMemory managedMemory
+}
+
+func newMemoryUsage() memoryUsage {
+	usage := memoryUsage{
+		appUsage:    map[string]managedMemory{},
+		totalMemory: managedMemory{label: totalKey},
+	}
+	return usage
```

Comment:
nit: usage variable is not used.

```suggestion
	return memoryUsage{
		appUsage:    map[string]managedMemory{},
		totalMemory: managedMemory{label: totalKey},
	}
```